src/sentry/sentry_metrics/querying/data_v2/execution.py

```diff
@@ -99,160 +96,221 @@ def _build_aligned_seq(
     return aligned_seq
 
 
-@dataclass(frozen=True)
-class ExecutableQuery:
-    with_series: bool
-    with_totals: bool
+def _push_down_group_filters(
+    metrics_query: MetricsQuery, groups_collection: GroupsCollection | None
+) -> MetricsQuery:
+    """
+    Returns a new `MetricsQuery` with a series of filters that ensure that the new query will have the same
+    groups returned. Keep in mind that there is no guarantee that all the groups will be returned, since data might
+    change in the meanwhile, so the guarantee of this method is that the returned groups will all be belonging to
+    `groups_collection`.
+
+    The need for this filter arises because when executing multiple queries, we want to have the same groups
+    returned, in order to make results consistent. Note that in case queries have different groups, some results
+    might be missing, since the reference query dictates which values are returned during the alignment process.
+    """
+    if not groups_collection:
+        return metrics_query
+
+    # We perform a transformation in the form [(key_1 = value_1 AND key_2 = value_2) OR (key_3 = value_3)].
+    groups_filters = []
+    for groups in groups_collection:
+        inner_snuba_filters = []
+        for filter_key, filter_value in groups:
+            inner_snuba_filters.append(Condition(Column(filter_key), Op.EQ, filter_value))
+
+        # In case we have more than one filter, we have to group them into an `AND`.
+        if len(inner_snuba_filters) > 1:
+            groups_filters.append(BooleanCondition(BooleanOp.AND, inner_snuba_filters))
+        else:
+            groups_filters.append(inner_snuba_filters[0])
+
+    # In case we have more than one filter, we have to group them into an `OR`.
+    if len(groups_filters) > 1:
+        groups_filters = [BooleanCondition(BooleanOp.OR, groups_filters)]
 
-    identifier: str
+    merged_query = TimeseriesConditionInjectionVisitor(groups_filters).visit(metrics_query.query)
+    return metrics_query.set_query(merged_query)
+
+
+class ScheduledQueryType(Enum):
+    SERIES = 0
+    TOTALS = 1
+
+
+@dataclass(frozen=True)
+class ScheduledQuery:
+    type: ScheduledQueryType
     metrics_query: MetricsQuery
+    next: Union["ScheduledQuery", None]
     order: QueryOrder | None
     limit: int | None
 
-    def is_empty(self) -> bool:
-        return not self.metrics_query.scope.org_ids or not self.metrics_query.scope.project_ids
+    def initialize(
```

Comment:
In the new implementation, I switch completely from having queries being resolved before being run to queries being resolved before they are added to the `_scheduled_queries` list. This allows us to make the code simpler since we assume that the queries are already correctly setup based on their type.